service/controller/app/resource/chart/desired.go

```diff
@@ -111,6 +110,124 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	return chartCR, nil
 }
 
+func (r *Resource) pickRepositoryURL(ctx context.Context, cc *controllercontext.Context, cr v1alpha1.App, chartName string) (string, error) {
+	if len(cc.Catalog.Spec.Repositories) == 1 {
+		return cc.Catalog.Spec.Repositories[0].URL, nil
+	}
+
+	var chart v1alpha1.Chart
+	err := cc.Clients.K8s.CtrlClient().Get(
+		ctx,
+		types.NamespacedName{Name: chartName, Namespace: r.chartNamespace},
+		&chart,
+	)
+	if apierrors.IsNotFound(err) || tenant.IsAPINotAvailable(err) {
+		// Repositories is guaranteed by Custom Resource Definition to have at least one entry.
+		return cc.Catalog.Spec.Repositories[0].URL, nil
+	} else if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// Check currently selected repository
+	repositoryIndex := -1
+	for i, repo := range cc.Catalog.Spec.Repositories {
+		if strings.Contains(chart.Spec.TarballURL, repo.URL) {
+			repositoryIndex = i
+			break
+		}
+	}
+	if repositoryIndex == -1 {
+		// Could not match current tarballURL to any of Catalog's repositories.
+		// Maybe the list was updated. Let's pick any existing repository.
+		r.logger.Debugf(ctx, "could not match tarball URL %q to any of %q Catalog repositories; using default", chart.Spec.TarballURL, cc.Catalog.Name)
+		return cc.Catalog.Spec.Repositories[0].URL, nil
+	}
+
+	if chart.Status.Release.Status == chartPullFailedStatus {
+		// chart-operator had trouble pulling the chart -- this includes timeouts and chart not being found (404)
+		// Round-robin the repository.
+		repositoryIndex = repositoryIndex + 1%len(cc.Catalog.Spec.Repositories)
```

Comment:
```suggestion
		repositoryIndex = repositoryIndex + 1 % len(cc.Catalog.Spec.Repositories)
```

Reply:
I fixed it anyways, it was missing parentheses.